### DISCUSSION TITLE
how to give subject ID & how to allocate Quota

### DISCUSSION THREAD

**Discussion by Jina-Kim-1108** [jspsych/jsPsych#879] [Q&A] [answered]
I am very new to programming code and jspsych is super helpful! Thanks so much for your tutorials.

I am currently working on modifying 'cloze plug-in' and I have a few questions here.

1. I would like to give subject ID
2. My study has a quota of gender and age. So if 50 males are done with study, 51st male should not start the study. How can I program of it?  
3. For the

**Comment by conbainbridge** [comment, marked as answer]
1. One way you could do this is to set up a random number generator ([a few ideas here](https://dev.to/rahmanfadhil/how-to-generate-unique-id-in-javascript-1b13)) to build unique IDs in the experiment file. You will probably want to store and handle these ideas in the backend.

2. This will require having some kind of backend setup, and the relevant code would likely live in your backend code rather than in the jsPsych experiment code (see discussion [#815](https://github.com/jspsych/jsPsych/discussions/815)). Session 4 of the workshop will look at [Pushkin](https://languagelearninglab.gitbook.io/pushkin/), one option for getting setup to handle data and backend development.

What was the rest of your third question going to be?

**Reply by jodeleeuw** [reply]
jsPsych also has a built in [random ID](https://www.jspsych.org/core_library/jspsych-randomization/#jspsychrandomizationrandomid) function that is very likely to generate unique IDs.

```js
console.log(jsPsych.randomization.randomID());
// outputs: "t7dwz0e713pc8juuaayyfvpkdd9un239"

console.log(jsPsych.randomization.randomID(8));
// outputs: "3xtpcbck"
```

**Reply by Jina-Kim-1108** [reply]
Thank you for all your help!
